src/Parallel/ParallelAnalyser.php

```diff
@@ -221,6 +227,7 @@ public function analyse(
 			$internalErrorsCount === 0 ? $dependencies : null,
 			$exportedNodes,
 			$reachedInternalErrorsCountLimit,
+			(int) array_sum($peakMemoryUsages), // not 100% correct as the peak usages of workers might not have met
```

Comment:
I dont think there is a better option. Still imo much better than before. It seems working well when comparing with system memory used.

Reply:
Tested against linux `ps` and it gives very good results, almost equal.